File: 20A40/cir-agus/EJER_21/Lectura de archivos/Articulo.cpp
# include <iostream>
# include <string>
# include "Articulo.h"

using namespace std;

Articulo::Articulo(){}

Articulo::Articulo(string newDes, string newUni, double newCanti, double newPre)
{
    this -> descripcion = newDes;
    this -> unidad = newUni;
    this -> cantidad = newCanti;
    this -> precio = newPre;
}

bool Articulo::setDescripcion(string d)
{
    if(d.length() > 0)
    {
        descripcion = d;
        return true;

    } else return false;
}

bool Articulo::setUnidad(string u)
{
    if(u.length() > 0)
    {
        unidad = u;
        return true;
    }else return false;
}

bool Articulo::setCantidad(double c)
{
    if(c > 0)
    {
        cantidad = c;
        return true;
    } else return false;
}

bool Articulo::setPrecio(double p)
{
    if(p > 0)
    {
        precio = p;
        return true;
    } else return false;
}

string Articulo::getDescripcion(){return descripcion;}
string Articulo::getUnidad(){return unidad;}
double Articulo::getCantidad(){return cantidad;}
double Articulo::getPrecio(){return precio;}

string Articulo::toString()
{
    string cadena = "Descripcio: "+descripcion;
    cadena += "\nUnidad: "+unidad;
    cadena += "\nCantidad: "+to_string(cantidad);
    cadena += "\nPrecio: "+to_string(precio);
    return cadena;
}

File: 20A40/cir-Israel/Ejercicio 21/Articulo.java
public class Articulo {
    private String Descripcion;
    private String Unidad;
    private double PrecioU;
    private double Cantidad;
    
    public boolean setDescripcion(String Des) {
        if (Des.length() > 0) {
            Descripcion = Des;
            return true;
        } else 
            return false;
    }

    public boolean setUnidad(String Uni) {
        if (Uni.length() > 0) {
            Unidad = Uni;
            return true;
        } else 
            return false;
    }

    public boolean setPrecioU(double PU) {
        if (PU > 0) {
            PrecioU = PU;
            return true;
        } else 
            return false;
    }

    public boolean setCantidad(double Cant) {
        if (Cant > 0) {
            Cantidad = Cant;
            return true;
        } else 
            return false;
    }

    public String getDescripcion() { return Descripcion; }
    public String getUnidad() { return Unidad; }
    public double getPrecioU() { return PrecioU; }
    public double getCantidad() { return Cantidad; }

    public String toString() {
        String resultado = "Datos";
        resultado += " Descripcion: " + Descripcion;
        resultado += ", Unidad: " + Unidad;
        resultado += ", PrecioU: " + PrecioU;
        resultado += ", Cantidad: " + Cantidad;
        return resultado;
    }
}